### PR TITLE
Make Logger accept only Target instances

### PR DESCRIPTION
**Pull request by machour**
Fixed issue #11 by accepting only `Target` instances for Logger.

### CODE REVIEW

File: src/Logger.php [hunk 1]
```diff
@@ -129,26 +117,22 @@ public function getTarget($name): ?Target
     }
 
     /**
-     * @param array|Target[] $targets the log targets. Each array element represents a single [[Target|log target]] instance
+     * @param Target[] $targets the log targets. Each array element represents a single [[Target|log target]] instance
      * or the configuration for creating the log target instance.
      */
     public function setTargets(array $targets): void
```
Comment (machour) [inline src/Logger.php]: Since I cannot typehint as `Target[]` in PHP, should I manually check the passed parameters?

Comment (samdark) [reply src/Logger.php]: Either that or we can implement something like:

```php

class Targets implements \IteratorAggregate
{
    /**
     * @var Target[]
     */
    private $targets = [];

    public function add(Target $target): self
    {
        $new = clone $this;
        $new->targets[] = $target;
        return $new;
    }

    public function getIterator()
    {
        return new \ArrayIterator($this->targets);
    }
}
```

Comment (rob006) [reply src/Logger.php]: > Since I cannot typehint as `Target[]` in PHP

We could also implement this as `setTargets(Target ...$targets)`.

Comment (machour) [reply src/Logger.php]: Thank you both. Took the simple path for now, will revise if we standardize instances arrays.

**Pull request merged**